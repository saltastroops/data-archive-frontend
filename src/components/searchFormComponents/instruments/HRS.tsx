import * as React from "react";
import { HRSMode, IHRS } from "../../../utils/ObservationQueryParameters";
import { MainGrid, SubGrid } from "../../basicComponents/Grids";
import SelectField, { AnyOption } from "../../basicComponents/SelectField";

interface IHrsProps {
  hrs: IHRS;
  onChange: (value: any) => void;
}

/**
 * A form for selecting HRS-related search parameters.
 * @param props
 * @constructor
 */
const HRS = (props: IHrsProps) => {
  const { onChange, hrs } = props;

  // Function for handling change events
  const change = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const name = e.target.name;
    const value = e.target.value;
    onChange({
      ...hrs,
      [name]: value
    });
  };

  const modes: HRSMode[] = [
    "High Resolution",
    "High Stability",
    "Int Cal Fibre",
    "Low Resolution",
    "Medium Resolution"
  ];

  return (
    <MainGrid>
      <SubGrid>
        <p>Mode</p>
        <SelectField name={"mode"} value={hrs.mode} onChange={change}>
          <AnyOption />
          {modes.map(mode => (
            <option key={mode} value={mode.toUpperCase()}>
              {mode}
            </option>
          ))}
        </SelectField>
      </SubGrid>
    </MainGrid>
  );
};
export default HRS;
