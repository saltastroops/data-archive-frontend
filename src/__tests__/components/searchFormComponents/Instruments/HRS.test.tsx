import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import HRS from "../../../../components/searchFormComponents/instruments/HRS";

const onChange = jest.fn();
describe("HRS", () => {
  it("should be defined", () => {
    expect(
      mount(<HRS hrs={{ errors: {}, name: "HRS" }} onChange={onChange} />)
    ).toBeDefined();
  });

  it("should render correctly without a mode", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        mount(<HRS hrs={{ errors: {}, name: "HRS" }} onChange={onChange} />)
      )
    ).toMatchSnapshot();
  });

  it("should render correctly with a mode", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(
        mount(
          <HRS
            hrs={{
              errors: {},
              filter: "H-alpha",
              mode: "Imaging",
              name: "HRS"
            }}
            onChange={onChange}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
