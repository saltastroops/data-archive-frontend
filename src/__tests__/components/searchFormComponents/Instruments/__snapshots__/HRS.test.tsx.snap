// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HRS should render correctly with a mode 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<HRS
  hrs={
    Object {
      "errors": Object {},
      "filter": "H-alpha",
      "mode": "Imaging",
      "name": "HRS",
    }
  }
  onChange={[MockFunction]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Mode
              </p>
              <SelectField
                name="mode"
                onChange={[Function]}
                value="Imaging"
              >
                <div
                  className="control select "
                >
                  <select
                    name="mode"
                    onChange={[Function]}
                    value="Imaging"
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="High Resolution"
                      value="HIGH RESOLUTION"
                    >
                      High Resolution
                    </option>
                    <option
                      key="High Stability"
                      value="HIGH STABILITY"
                    >
                      High Stability
                    </option>
                    <option
                      key="Int Cal Fibre"
                      value="INT CAL FIBRE"
                    >
                      Int Cal Fibre
                    </option>
                    <option
                      key="Low Resolution"
                      value="LOW RESOLUTION"
                    >
                      Low Resolution
                    </option>
                    <option
                      key="Medium Resolution"
                      value="MEDIUM RESOLUTION"
                    >
                      Medium Resolution
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</HRS>
`;

exports[`HRS should render correctly without a mode 1`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

.c1.c1 {
  text-align: left;
  padding-left: 10px;
}

<HRS
  hrs={
    Object {
      "errors": Object {},
      "name": "HRS",
    }
  }
  onChange={[MockFunction]}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "main-grid",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="main-grid c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "sub-grid",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sub-grid c1"
            >
              <p>
                Mode
              </p>
              <SelectField
                name="mode"
                onChange={[Function]}
              >
                <div
                  className="control select "
                >
                  <select
                    name="mode"
                    onChange={[Function]}
                  >
                    <AnyOption>
                      <option
                        key="any"
                        value=""
                      >
                        Any
                      </option>
                    </AnyOption>
                    <option
                      key="High Resolution"
                      value="HIGH RESOLUTION"
                    >
                      High Resolution
                    </option>
                    <option
                      key="High Stability"
                      value="HIGH STABILITY"
                    >
                      High Stability
                    </option>
                    <option
                      key="Int Cal Fibre"
                      value="INT CAL FIBRE"
                    >
                      Int Cal Fibre
                    </option>
                    <option
                      key="Low Resolution"
                      value="LOW RESOLUTION"
                    >
                      Low Resolution
                    </option>
                    <option
                      key="Medium Resolution"
                      value="MEDIUM RESOLUTION"
                    >
                      Medium Resolution
                    </option>
                  </select>
                </div>
              </SelectField>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</HRS>
`;
